library/ix-dev/community/ipfs/templates/_ipfs.tpl

```diff
@@ -0,0 +1,149 @@
+{{- define "ipfs.workload" -}}
+workload:
+  ipfs:
+    enabled: true
+    primary: true
+    type: Deployment
+    podSpec:
+      hostNetwork: {{ .Values.ipfsNetwork.hostNetwork }}
+      containers:
+        ipfs:
+          enabled: true
+          primary: true
+          imageSelector: image
+          securityContext:
+            runAsUser: {{ .Values.ipfsRunAs.user }}
+            runAsGroup: {{ .Values.ipfsRunAs.group }}
+          {{ with .Values.ipfsConfig.additionalEnvs }}
+          env:
+            {{ range $env := . }}
+            {{ $env.name }}: {{ $env.value }}
+            {{ end }}
+          {{ end }}
+          probes:
+            liveness:
+              enabled: true
+              type: exec
+              command:
+                - ipfs
+                - dag
+                - stat
+                # https://github.com/ipfs/kubo/blob/8f638dcbcd875ecff92021e4b62d0af8848022ce/Dockerfile#L116
+                - /ipfs/QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn
+            readiness:
+              enabled: true
+              type: exec
+              command:
+                - ipfs
+                - dag
+                - stat
+                - /ipfs/QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn
+            startup:
+              enabled: true
+              type: exec
+              command:
+                - ipfs
+                - dag
+                - stat
+                - /ipfs/QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn
+      initContainers:
+      {{- include "ix.v1.common.app.permissions" (dict "containerName" "01-permissions"
+                                                        "UID" .Values.ipfsRunAs.user
+                                                        "GID" .Values.ipfsRunAs.group
+                                                        "type" "install") | nindent 8 }}
```

Comment:
Do you think having another workload which only targets the install time workload makes more sense?
It feels weird that we have a workload defined which will be actually running ipfs and then we have under init container some metadata which says that this is only applicable on installatino of chart?

Reply:
So the `01-permissions` container is marked as `type: install`. Which common puts it between an if block of `.Release.IsInstall`.
Meaning it will only render at installation, upgrades will not render this at all.
ipfs container does not run as root, so it won't be able to do any chown.

The `02-init-config` is `type: init`, meaning it will get rendered (and run on every start of the app).
Which in this case is needed as it configures the ports.

Reply:
I see..i was seeing this differently as being a pre-install job or something but obviously that is not the case

Reply:
Nop, it's just containers in the same Deployment.

Opted to not go for jobs in this case, as it's quick actions. And spinning a whole extra pod would require more configuration